### PR TITLE
refactor: use Auth V2 API

### PR DESCRIPTION
**Pull request by amishshah**
This performs a minimal update to the HS Discord API allowing it to work with Auth v2 API. The only thing that I can think of that it is currently missing is the sponsors role.

### CODE REVIEW

File: src/controllers/discord/OAuth2Controller.ts [hunk 1]
```diff
@@ -101,15 +108,18 @@ export class OAuth2Controller {
 		return this.rest.patch(`/guilds/${this.api.options.discord.guildId}/members/${userId}`, data);
 	}
 
-	private async getAuthRole(level: AuthLevel): Promise<DiscordResource> {
-		if (level === AuthLevel.Organiser) {
+	private async getAuthRole(user: User): Promise<DiscordResource> {
+		// todo: once the auth client is updated, pass the user's id in as a parameter here.
```
Comment (kzalys) [inline src/controllers/discord/OAuth2Controller.ts]:
```suggestion
```

File: src/controllers/discord/OAuth2Controller.ts [hunk 2]
```diff
@@ -20,6 +21,12 @@ interface TokenResponse {
 	scope: string;
 }
 
+enum AuthRole {
```
Comment (kzalys) [inline src/controllers/discord/OAuth2Controller.ts]: These are URIs for discord roles, right?

**Review by kzalys — CHANGES_REQUESTED** [review]

File: src/controllers/discord/OAuth2Controller.ts (hunk 2 above)
Comment (amishshah) [reply src/controllers/discord/OAuth2Controller.ts]: Yep, I guess the enum could be named more clearly. Will update this tomorrow.

Comment (kzalys) [reply src/controllers/discord/OAuth2Controller.ts]: Could you rename this to something like `DiscordRolesUri`. `AuthRole` implies that you're referring to the roles on `hs_auth`

**Review by kzalys — APPROVED** [review]

**Pull request merged**